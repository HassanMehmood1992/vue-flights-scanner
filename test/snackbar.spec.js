import { mount, createLocalVue } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import Vuetify from "vuetify";
import Vuex from "vuex";

import snackbar from "@/components/snackbar";
import defaultLayout from "@/layouts/default";

const localVue = createLocalVue();
localVue.use(Vuex);

let wrapper;

beforeEach(() => {
  let vuetify = new Vuetify();

  wrapper = mount(defaultLayout, {
    store: new Vuex.Store({
      state: { products: [] }
    }),
    localVue,
    vuetify,
    stubs: {
      NuxtLink: true,
      Nuxt: true
      // Any other component that you want stubbed
    }
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe("Snackbar component", () => {
  test("should generate snackbar", async () => {
    // With jest we can create snapshot files of the HTML output
    // expect(wrapper.html()).toMatchSnapshot()
    const testWrapper = shallowMount(snackbar);
    expect(testWrapper.element).toMatchSnapshot();
  });
});
