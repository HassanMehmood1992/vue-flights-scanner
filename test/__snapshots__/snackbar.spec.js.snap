// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar component should generate snackbar 1`] = `
<v-snackbar-stub
  contentclass=""
  right="true"
  tag="div"
  timeout="5000"
  top="true"
  transition="v-snack-transition"
>
  
    
    
  <v-btn-stub
    activeclass=""
    dark="true"
    icon="true"
    small="true"
    tag="button"
    text="true"
    type="button"
  >
    <v-icon-stub
      small=""
    >
      mdi-close
    </v-icon-stub>
  </v-btn-stub>
</v-snackbar-stub>
`;
